src/decode_metar.c

```diff
@@ -1067,7 +1067,28 @@ static float prevailingVisibility( char *visibility )
    }
  
 }
- 
+
+/*
+ * Occasionally, visibility strings are preceded by "M ", indicating
+ * that the actual visibility is something less than the value
+ * in question, eg, M 1/4SM = 'less than 0.25 sm'
+ */
+static MDSP_BOOL isMBeforeVis( char **m, Decoded_METAR *Mptr,
+                               int *NDEX )
+{
+    (void)Mptr;
+
+    if (*m == NULL) {
+        return FALSE;
+    }
+
+    if (strncmp(*m, "M ", 2) == 0) {
```

Comment:
Yes -- the "no space" case is also addressed (https://github.com/agbrooks/mdsplib/blob/master/src/decode_metar.c#L1135). While it would be nice to handle them in the same place in the code, if the M is separated by a space, it ends up in its own token before rest of the visibility string.

Reply:
Maybe a comment to indicate that the "no space" case is handled elsewhere?

Reply:
good idea -- I'll add one